File: scala/finatra-demo/src/main/scala/com/example/jpa/AnotherThing.scala
package com.example.jpa

import javax.persistence.{Column, Entity, Table}

import scala.beans.BeanProperty

@Entity
@Table(name = "t_another_thing")
class AnotherThing extends BaseEntity {
  @BeanProperty
  @Column
  var name: String = _
  @BeanProperty
  @Column
  var flag: String = _
}

object AnotherThing {
  def apply(id: Long, name: String, flag: String): AnotherThing = {
    val thing = new AnotherThing()
    thing.id = id
    thing.name = name
    thing.flag = flag
    thing
  }
}

File: scala/finatra-demo/src/main/java/com/example/jpa/AnotherThing.java
package com.example.jpa;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "t_another_thing")
public class AnotherThing extends BaseEntity {
    @Column
    public String name;
    @Column
    public String flag;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getFlag() {
        return flag;
    }

    public void setFlag(String flag) {
        this.flag = flag;
    }
}
